### PR TITLE
Replace RAML with Swagger and removed ABAO tests

### PR DESCRIPTION
**Pull request by ehlertjd**
Please see the Swagger [README](https://github.com/scitran/core/blob/swagger/swagger/README.md) and [STYLE_GUIDE](https://github.com/scitran/core/blob/swagger/swagger/STYLE_GUIDE.md) for the basics of running and writing Swagger documentation. 

You can also see a list of undocumented APIs by first running the integration tests, then running (from the swagger folder):
```
> grunt coverage
```

This is ready for review, but not quite ready to merge until we resolve where the generated API Docs for master are going to live.

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by ehlertjd** [comment]
@ryansanford Swagger 2.0 does not support specification for multiple file upload, so I will need to add an extension for code-generation to support those methods in the generated code. I will update the affected endpoints once an approach has been determined.

**Comment by ehlertjd** [comment]
@nagem @ambrussimon This has been rebased on master and is ready for review, thank you!

**Comment by ehlertjd** [comment]
@ryansanford @gsfr The devops changes are ready for review.

I moved the github setup code into `.travis.yml` and pass any relevant variables into `bin/push-docs.sh` which does the push/prune of `gh-pages`. This script can now be run independently of CI.

**Comment by ryansanford** [comment]
Looks like the tag build worked as expected. https://scitran.github.io/core/tags/swagger-pr-tag-test/

**Comment by ehlertjd** [comment]
@ryansanford Pretty confident about the `master` branch code path as well as the conflict resolution for branches/tags. I hand-tested a bunch of scenarios using a local upstream while developing the script.

I had thought about the branch documentation - do we maybe want to generate an index page for the branches/ tags/ subdirectories since github hosting doesn't seem to do that for us? If we did that, I could add those as sub-links under the main Documentation link.

**Comment by ryansanford** [comment]
I don't want to keep blocking this PR with scope creep, like generation and maintenance of an index page. I'd be happy with just a note in the README.md

**Comment by ehlertjd** [comment]
@ryansanford Note added to README.md

**Comment by nagem** [comment]
I think all the feedback I've had has been given verbally over the course of this project - at this point, LGTM! I've passed over the new Swagger docs to look for any abnormalities a couple of times but haven't spotted any. Doesn't mean we're clear of them, but I'd also like to go through and up the quality of some of the original docs. I'm sure any inconsistencies we missed can be found then. If it passes tests and passes a best-effort visual review, I'm good to go.

**Comment by nagem** [comment]
I still have to commit the docs I wrote for some report endpoints. Will mark my review as complete but wanted to note those are coming soon.

File: README.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ SciTran Core is a RESTful HTTP API, written in Python and backed by MongoDB. It
 
 ### [Documentation](https://scitran.github.io/core)
 
+API documentation for branches and tags can be found at https://scitran.github.io/branches/**branchname** and
+https://scitran.github.io/tags/**tagname**.
+
```
Comment (gsfr) [inline README.md]: The links are missing a `/core` particle and the `**` renders weird. Maybe try angle brackets?

**Pull request merged**